### DISCUSSION TITLE
Im feeling dumb but need the help

### DISCUSSION THREAD

**Discussion by jayket** [codenotary/immudb#1596] [Q&A] [answered]
Long story short im trying to build a journal that cant be edited. I was having issues getting immudb running on a pi4b arm32 i elected to redo everything to arm64 and i was able to build the docker images for immudb&immuclient and have them running from portainer but i cant seem to exec into the container or have anything connect to the container to pull the data or add the data. what am i doing wrong? I have attached the logs and the compose file im using. Any help of where im going wrong would be useful.
  
`` immudb 1.4.1
Commit  : be85a0e20ba43670d93ed2694da8df7350ca30e6
Built at: Sat, 01 Apr 2023 14:46:44 UTC
Static  : true
================ Config ================
Data dir         : /var/lib/immudb
Address          : 0.0.0.0:3322
Metrics address  : 0.0.0.0:9497/metrics
Sync replication : false
Config file      : configs/immudb.toml
Log format       : text
Max recv msg size: 33554432
Auth enabled     : true
Dev mode         : true
Default database : defaultdb
Maintenance mode : false
Synced mode      : true

Superadmin default credentials
   Username      : immudb
   Password      : immudb

immudb 2023/04/01 16:21:28 INFO: Creating database 'systemdb' {replica = false}...
immudb 2023/04/01 16:21:28 INFO: Binary Linking up to date at '/var/lib/immudb/systemdb'
immudb 2023/04/01 16:21:28 INFO: Index '/var/lib/immudb/systemdb/index' {ts=0, discarded_snapshots=0} successfully loaded
immudb 2023/04/01 16:21:28 INFO: Indexing in progress at '/var/lib/immudb/systemdb'
immudb 2023/04/01 16:21:28 INFO: Loading SQL Engine for database 'systemdb' {replica = false}...
immudb 2023/04/01 16:21:28 INFO: SQL Engine ready for database 'systemdb' {replica = false}
immudb 2023/04/01 16:21:28 INFO: Database 'systemdb' successfully created {replica = false}
immudb 2023/04/01 16:21:29 INFO: Admin user 'immudb' successfully created
immudb 2023/04/01 16:21:29 INFO: Creating database 'defaultdb' {replica = false}...
immudb 2023/04/01 16:21:29 INFO: Binary Linking up to date at '/var/lib/immudb/defaultdb'
immudb 2023/04/01 16:21:29 INFO: Index '/var/lib/immudb/defaultdb/index' {ts=0, discarded_snapshots=0} successfully loaded
immudb 2023/04/01 16:21:29 INFO: Indexing in progress at '/var/lib/immudb/defaultdb'
immudb 2023/04/01 16:21:29 INFO: Loading SQL Engine for database 'defaultdb' {replica = false}...
immudb 2023/04/01 16:21:29 INFO: SQL Engine ready for database 'defaultdb' {replica = false}
immudb 2023/04/01 16:21:29 INFO: Database 'defaultdb' successfully created {replica = false}
immudb 2023/04/01 16:21:29 INFO: Started with an empty default database
immudb 2023/04/01 16:21:29 INFO: Flushing index '/var/lib/immudb/systemdb/index' {ts=1, cleanup_percentage=0.00/0.00, since_cleanup=1} requested via SnapshotSince...
immudb 2023/04/01 16:21:29 INFO: Index '/var/lib/immudb/systemdb/index' {ts=1, cleanup_percentage=0.00/0.00} successfully flushed
immudb 2023/04/01 16:21:29 INFO: Flushing index '/var/lib/immudb/systemdb/index' {ts=1} finished with: 1 inner nodes, 0 leaf nodes, 1 entries
immudb 2023/04/01 16:21:29 INFO: sessions guard started
immudb 2023/04/01 16:21:29 INFO: pgsql server is running at port 5432
immudb 2023/04/01 16:21:29 INFO: Webconsole enabled: 0.0.0.0:8080
immudb 2023/04/01 16:21:29 INFO: Web API server enabled on 0.0.0.0:8080/api (http)
You can now use immuadmin and immuclient CLIs to login with the immudb superadmin user and start using immudb. ``

`version: "3"
services:
  immuclient:
        container_name: Immudb_Client
        volumes:
            - '/home/test/immudb/:/data/'
        image: 'immuclient:latest'
        depends_on:``
             - immudb
   immudb:
        container_name: Immudb
        ports:
            - 3322:3322
            - 9497:9497
            - 5432:5432
        volumes:
            - '/home/test/immudb/:/data/'
        image: 'immudb:latest' `

**Comment by jayket** [comment]
Im trying to connect this to budibase for a front end to add the data once i have it set up.

**Comment by Razikus** [comment]
Hello

Correct me if I'm wrong - you are trying to use BudyBase Postgres integration directtly with 5432 psql port of immudb, right?

**Reply by jayket** [reply]
Yes that’s correct. I had issues setting up immudb prior like I said but I was able to build the immudb and client images locally each running in their own container. When I use http://localhost:5432 I do get a webpage showing (in Firefox)“
![image](https://user-images.githubusercontent.com/108714822/229319936-4f10efb7-31c1-402e-9f4f-5f00557a2e78.jpg)
” 
I honestly just don’t know what side the issue is on though. And I want to make sure I resolved all my issues with immudb first

**Comment by Razikus** [comment, marked as answer]
I think immudb is working correctly. Your issue is that postgresql layer is compatible only on communication level not on syntax level - that means that immudb supports right now part or postgres layer
The best is to use dedicated immudb SDK to communicate with immudb 

Budybase will not work now, maybe it would be cool to implement plugin for it

**Reply by jayket** [reply]
Thank you for all the help and feedback I was just being hopeful there was a easy way to use this db without building a front end from scratch because coding isn’t my forte. Thanks again for the help.

**Reply by jeroiraz** [reply]
> Thank you for all the help and feedback I was just being hopeful there was a easy way to use this db without building a front end from scratch because coding isn’t my forte. Thanks again for the help.

Something that might help, have you played a little bit with the embedded webconsole (https://docs.immudb.io/master/connecting/webconsole.html) ?, if you are coding in go, you may also use the `database/sql` package (https://docs.immudb.io/master/develop/sql/sqlstdlib.html)
